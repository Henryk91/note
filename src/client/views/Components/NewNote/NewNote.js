/* eslint-disable no-plusplus */
/* eslint-disable react/sort-comp */
/* eslint-disable react/no-unused-state */
/* eslint-disable no-use-before-define */
/* eslint-disable no-unused-expressions */
/* eslint-disable import/order */
/* eslint-disable react/prop-types */
/* eslint-disable arrow-parens */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { EditNoteCheck } from '../index';
import { Link } from 'react-router-dom';

export default class NewNote extends Component {
  constructor(props) {
    super(props);
    this.state = {
      radioType: 'Number'
    };
    this.addNewUser = this.addNewUser.bind(this);
    this.setRadioType = this.setRadioType.bind(this);
  }

  addNewUser = event => {
    event.preventDefault();
    const heading = event.target.heading.value;
    let number = event.target.number.value;
    let tag = event.target.tagType.value;

    tag === 'Note' ? (tag = event.target.tagTypeText.value) : tag;
    const loginKey = localStorage.getItem('loginKey');
    const uniqueId = docId();
    const textTag = event.target.tagTypeText.value;
    if (tag === 'Log') {
      number = JSON.stringify({ json: true, date: textTag, data: number });
    }

    const note = {
      id: uniqueId,
      userId: loginKey,
      createdBy: 'Unknown',
      heading,
      dataLable: [{ tag, data: number }]
    };
    this.setState({ showAddItem: false });
    this.props.set({ note });
    window.history.back()
  };

  setRadioType(type) {
    this.setState({ radioType: type });
  }

  render() {
    const { Theme } = this.props
    const themeBack = `${Theme.toLowerCase()}-back`;
    const themeHover = `${Theme.toLowerCase()}-hover`;
    
    return (
      <div>
        <Link className="backButton" style={{ textDecoration: 'none' }} to="/" title="Note List">
          <i className="fas fa-arrow-left" />
        </Link>
        <form onSubmit={this.addNewUser}>
          <br />
          <input className={themeBack}  name="heading" type="text" placeholder="Heading" required="required" />
          <br />
          <EditNoteCheck Theme={this.props.Theme} />
          <button className={`submit-button ${themeHover} ${themeBack}`} type="submit">
            {' '}
            <i className="fas fa-check" />
          </button>
        </form>
      </div>
    );
  }
}

const docId = () => {
  let text = '';

  const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';

  for (let i = 0; i < 20; i++) {
    text += possible.charAt(Math.floor(Math.random() * possible.length));
  }
  return text;
};
