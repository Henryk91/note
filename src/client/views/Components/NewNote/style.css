
input[type="radio"]{
    width: 50px;
}
label{
    margin: 5px;
}