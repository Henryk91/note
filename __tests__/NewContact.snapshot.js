
import NewNote from '../src/client/views/Components/NewNote/NewNote';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

test('NoteDetail component renders all details', () => {

    const rendered = renderer.create(
        <Router>
            <NewNote />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});