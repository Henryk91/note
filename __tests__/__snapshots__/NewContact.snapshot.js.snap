// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetail component renders all details 1`] = `
<form
  onSubmit={[Function]}
>
  <br />
  <input
    name="heading"
    placeholder="First Name"
    required="required"
    type="text"
  />
  <br />
  <input
    name="lastName"
    placeholder="last Name"
    required="required"
    type="text"
  />
  <br />
  <br />
  <div>
    <div
      className="radioBox"
    >
      <label>
        Number
      </label>
      <label>
        Email 
      </label>
      <label>
        Other
      </label>
      <br />
      <input
        defaultChecked={true}
        name="tagType"
        onClick={[Function]}
        type="radio"
        value="Number"
      />
      <input
        name="tagType"
        onClick={[Function]}
        type="radio"
        value="Email"
      />
      <input
        name="tagType"
        onClick={[Function]}
        type="radio"
        value="Other"
      />
    </div>
    <div>
      <input
        name="number"
        placeholder="Add Number"
        type="number"
      />
      <br />
      <br />
    </div>
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
